s3prl_vc/bin/train.py

```diff
@@ -213,19 +214,23 @@ def _train_step(self, batch):
             all_hs, all_hlens = self.upstream_model(xs, ilens)
         hs, hlens = self.upstream_featurizer(all_hs, all_hlens)
 
+        # initialize
+        gen_loss = 0.0
+
         # model forward
-        outs, outs_lens = self.model(hs, hlens, targets=ys, spk_embs=spembs, f0s=f0s)
+        predicted, targets, outs_lens = self.model(
+            hs, hlens, targets=ys, spk_embs=spembs, f0s=f0s
+        )
 
-        # normalize output
-        outs = (outs - self.config["trg_stats"]["mean"]) / self.config["trg_stats"][
-            "scale"
-        ]
-        ys = (ys - self.config["trg_stats"]["mean"]) / self.config["trg_stats"]["scale"]
+        if self.config["model_type"] == "Diffusion":
+            # diffusion model needs the inputs and outputs to be
+            # of the same length, so there are cases where it is
+            # cut to match with the upsampled upstream features
+            olens = outs_lens
```

Comment:
Hmm, can we put this in the loss required by the diffusion model? Please refer to this file: https://github.com/unilight/s3prl-vc/blob/main/s3prl_vc/losses/l1_loss.py#L25-L30

Reply:
Good idea, I moved it over to the L2 loss then.